File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
       vector < bool > ans(candies.size()) ;
       int Max = *max_element(candies.begin() , candies.end());
       for(int i = 0 ; i < candies.size() ; i++){
           if(candies[i] + extraCandies >= Max)
               ans[i] = true;
           else 
               ans[i] = false;
       }
        return ans ;
    }
    
};

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cs
public class Solution {
    public IList<bool> KidsWithCandies(int[] candies, int extraCandies) {
        int max = candies[0];
        for(int i= 1; i < candies.Length; i++){
            if(max < candies[i])
                max = candies[i];
        }

        var res = new List<bool>(candies.Length);
        for(int i = 0 ; i < candies.Length; i++){
            res.Add(candies[i] + extraCandies >= max);
        }
        return res;
    }
}